### PR TITLE
update: non-NodeView compatible Pubs

### PR DESCRIPTION
**Pull request by isTravis**
This PR implements the necessary changes so that `pubpub-editorv6.2.0` can be installed. It adds the necessary functionality to footnote and citations rendering to allow non-nodeview elements. It also makes necessary changes for image/video/iframe breakouts to properly render.

_Test Plan:_
- Pull down `plaindom` branch from `pubpub-editor`.
- Build and link `pubpub-editor` so that the non-nodeview editor code is used. 
- Verify that footnotes, citations, hovering, and server rendering are functional.
- Verify that other elements can be added to the editor.
- Add an image and set it's align option to breakout. Verify that it properly renders full-bleed on desktop and mobile.

_Unhandled by this PR, but necessary soon:_
- Styling of `<PubNoteContent />` component
- Re-design of Citation and Footnote input UI

### CODE REVIEW

File: client/containers/Pub/PubDocument/PubMouseEvents/PubMouseEvents.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	const timeouts = useRef({});
+	const [hoverElems, hoverElemsDispatch] = useReducer((state, action) => {
+		return {
+			...state,
+			[action.type]: action.elem,
+		};
+	}, {});
+
+	/* Specify the types of elems we want events for */
+	const mouseElemTypes = [
```
Comment (idreyn) [inline client/containers/Pub/PubDocument/PubMouseEvents/PubMouseEvents.js]: Break this constant out of the function body?

File: client/containers/Pub/PubDocument/PubMouseEvents/NotePopover.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import Popper from 'popper.js';
+import { Card } from '@blueprintjs/core';
+
+require('./notePopover.scss');
+
+const propTypes = {
+	elem: PropTypes.object.isRequired,
+	mainContentRef: PropTypes.object.isRequired,
+	timeouts: PropTypes.object.isRequired,
+	mouseLeave: PropTypes.func.isRequired,
```
Comment (idreyn) [inline client/containers/Pub/PubDocument/PubMouseEvents/NotePopover.js]: Since it's a callback, this prop should be named `onMouseLeave` for consistency's sake.

File: init.js [hunk 3]
```diff
@@ -17,12 +17,15 @@ throng(
 		lifetime: Infinity,
 	},
 	() => {
-		const { startServer } = require('./server/server.js');
+		const loadServer = () => {
```
Comment (idreyn) [inline init.js]: This should really be in a different commit.

File: client/containers/Pub/pub.scss [hunk 4]
```diff
@@ -44,14 +44,18 @@ $breakout-simplify-cutoff: $compact-cutoff-width / (1 - 2 * $pub-margin-fraction
...
-    .breakout {
-        width: 100vw !important;
-        margin-left: 0;
-        position: relative;
-        left: calc(-1 * (50vw - (100% + #{$gutter-full-width} + #{$comments-full-width}) / 2));
+    [data-align='breakout'] {
+        @for $i from 1 through 100 {
+            &[data-size='#{$i}'] {
+                width: $i * 1vw;
+                margin-left: 0;
```
Comment (idreyn) [inline client/containers/Pub/pub.scss]: I am worried about statically generating this much CSS in a `@for` loop for reasons we're discussing offline, but one small improvement we could make here is just moving the constant declarations, i.e.
```
margin-left: 0;
position: relative;
```
unless Sass has loop optimizations I'm not aware of, that should reduce the impact on the bundle size.

File: client/containers/Pub/PubDocument/PubBody.js [hunk 5]
```diff
@@ -88,6 +96,48 @@ const PubBody = (props) => {
 			onComplete(nextStatus);
 		}
 	};
+
+	useEffect(() => {
```
[the remark this answers is not in the document]
Comment (isTravis) [reply client/containers/Pub/PubDocument/PubBody.js]: This is great. I'm supportive of this direction, but for the sake of momentum, would like the address this refactor in a separate PR. Happy to tackle it if I get to it first, but also happy to have you implement what you have above. 

`/api/editor/citations-format` right now expects a single array of items to process, so we would need to refactor that a bit too. Or - we could keep track of the indexes on the client, push up a concatenated array, and then split the result when we get it back.

File: client/containers/Pub/pub.scss (hunk 4 above)
Comment (isTravis) [reply client/containers/Pub/pub.scss]: Handled in d6864b3.

File: client/containers/Pub/PubDocument/PubMouseEvents/NotePopover.js (hunk 2 above)
Comment (isTravis) [reply client/containers/Pub/PubDocument/PubMouseEvents/NotePopover.js]: Handled in d6864b3.

File: client/containers/Pub/PubDocument/PubMouseEvents/PubMouseEvents.js (hunk 1 above)
Comment (isTravis) [reply client/containers/Pub/PubDocument/PubMouseEvents/PubMouseEvents.js]: Handled in d6864b3.

File: client/components/FormattingBar/Controls/ControlsFootnote.js [hunk 6]
```diff
@@ -70,10 +58,15 @@ class ControlsFootnote extends Component {
 				<div className="block">
 					<div className="label">Structured Data Output</div>
 					<div className="input wide">
-						<div
-							className="rendered-citation"
-							dangerouslySetInnerHTML={{ __html: this.state.structuredHtml }}
-						/>
+						{hasRenderedContent && (
+							<PubNoteContent
+								structured={this.props.footnotes[this.props.attrs.count - 1].html}
```
[the remark this answers is not in the document]
Comment (isTravis) [reply client/components/FormattingBar/Controls/ControlsFootnote.js]: Handled in d6864b3.
